### PR TITLE
Add a small section on git submoduling the SDK

### PR DESCRIPTION
**Pull request by noahp**
Short reference for using the SDK as a git submodule.

### CODE REVIEW

File: docs/mcu/sdk-submodule-management.mdx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+By default, the submodule add command will checkout the latest commit on the
+default branch. To select a specific commit or tag for the submodule, follow
+these commands:
+
+```bash
+# change directory to the submodule
+cd path/to/memfault/submodule
+
+# checkout a particular tag
+git checkout 0.28.0
```
Comment (chrisc11) [inline docs/mcu/sdk-submodule-management.mdx:32]: Perhaps a callout that latest version can be found [here](https://github.com/memfault/memfault-firmware-sdk/releases) so folks don't pin to old tag

**Review by chrisc11 — APPROVED** [review]

File: docs/mcu/sdk-submodule-management.mdx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+# Option 2: clone, then update after
+git clone <repo remote URL>
+cd path-to-newly-cloned-repo
+git submodule update --init --recursive
+```
+
+## Reference
+
+https://github.blog/2016-02-01-working-with-submodules/
```
Comment (fnune) [inline docs/mcu/sdk-submodule-management.mdx:58]: Nit: We don't have a Reference section in any other doc AFAIK. Can we change this into a paragraph? E.g.

```suggestion
To learn more about working with submodules, check out [GitHub's blog post](https://github.blog/2016-02-01-working-with-submodules/) about them.
```

File: docs/mcu/sdk-submodule-management.mdx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+your build system doesn't natively support using Git Repositories as sources.
+
+To add the Memfault SDK as a Git Submodule:
+
+```bash
+# the last argument sets the path to the submodule, here "third_party/memfault-firmware-sdk"
+git submodule add https://github.com/memfault/memfault-firmware-sdk.git third_party/memfault-firmware-sdk
+```
+
+The submodule add is now staged in the git index, and can be committed as usual.
```
Comment (fnune) [inline docs/mcu/sdk-submodule-management.mdx:19]: Nit: you called it Git everywhere else in the doc.

```suggestion
The submodule add is now staged in the Git index, and can be committed as usual.
```

**Review by fnune — APPROVED** [review]

File: docs/mcu/sdk-submodule-management.mdx (hunk 1 above)
Comment (noahp) [reply docs/mcu/sdk-submodule-management.mdx]: good call! added

**Pull request merged**